patches/tModLoader/Terraria.ModLoader.Recipes/RecipeBuilder.cs

```diff
@@ -100,11 +112,17 @@ public RecipeBuilder Requires(params int[] types)
 		/// <exception cref="RecipeException">The item " + itemName + " does not exist in mod " + mod.Name + ". If you are trying to use a vanilla item, try removing the first argument.</exception>
 		public RecipeBuilder Requires(Mod mod, string itemName, int stack = 1)
 		{
-			_recipe.AddIngredient(mod ?? _recipe.mod, itemName, stack);
+			_recipe.AddIngredient(mod, itemName, stack);
 
 			return this;
 		}
 
+		/// <summary>Adds an ingredient to this recipe of the given type and stack size.</summary>
+		/// <typeparam name="T">The type.</typeparam>
+		/// <param name="stack">The stack.</param>
+		/// <returns></returns>
+		public RecipeBuilder Requires<T>(int stack = 1) where T : ModItem => Requires(ModContent.ItemType<T>(), stack);
```

Comment:
If we're on this route, you could make a `Requires(params Type[] types)` so you can do `Requires(typeof(MyItem))`
But it gets pretty verbose at that point, `Requires<T>` is fine